src/org/jetbrains/java/decompiler/modules/decompiler/sforms/SFormsConstructor.java

```diff
@@ -744,6 +785,54 @@ private SFormsFastMapDirect getFilteredOutMap(String nodeid, String predid, Dire
         }
 
         mapNew.intersection(mapNewTemp);
+//
+//        if (this.trackPhantomExitNodes && !mapTrueSource.isEmpty() && !mapNew.isEmpty()) { // FIXME: what for??
```

Comment:
Forgot to add this in the review, but you should add a TODO here saying that we need to implement this at a later date, as discussed